SwaggerGen/Swagger/Type/Custom/EmailType.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace SwaggerGen\Swagger\Type\Custom;
+
+/**
+ * Email type definition
+ *
+ * @package    SwaggerGen
+ * @author     Martijn van der Lee <martijn@vanderlee.com>
+ * @copyright  2014-2017 Martijn van der Lee
+ * @license    https://opensource.org/licenses/MIT MIT
+ */
+class EmailType extends \SwaggerGen\Swagger\Type\StringType implements ICustomType
+{
+
+	const PATTERN = '\A[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\z';
+	const TYPE = 'email';
+
+	public static function register($type = null)
+	{
+		self::registerType($type === null ? self::TYPE : $type, __CLASS__);
+	}
+
+	public static function unregister($type = null)
+	{
+		self::unregisterType($type === null ? self::TYPE : $type);
+	}
+
+	/**
+	 * Construct and setup the regular expression for this type
+	 * 
+	 * @param \SwaggerGen\Swagger\AbstractObject $parent
+	 * @param string $definition
+	 */
+	public function __construct(\SwaggerGen\Swagger\AbstractObject $parent, $definition)
+	{
+		$this->pattern = self::PATTERN;
+
+		parent::__construct($parent, $definition);
+	}
+
+	/**
+	 * Parse a type definition string, assuming it belongs to this type
+	 * 
+	 * @param string $definition
+	 * @throws \SwaggerGen\Exception
+	 */
+	protected function parseDefinition($definition)
+	{
+		$definition = self::trim($definition);
+
+		$match = array();
+		if (preg_match(self::REGEX_START . self::REGEX_FORMAT . self::REGEX_DEFAULT . self::REGEX_END, $definition, $match) !== 1) {
+			throw new \SwaggerGen\Exception("Unparseable email definition: '{$definition}'");
+		}
+
+		if (strtolower($match[1] !== self::TYPE)) {
```

Comment:
* This check would preclude using "type aliases" (?). 
* Also parentheses is messed up here (you're doing `strtolower($a !== $b)` instead of intended `strtolower($a) !== $b`)
* And it's, perhaps, better to use `strcasecmp()` anyway

Reply:
This applies to all custom types in this PR